src/expr/src/scalar/func/impls/timestamp.rs

```diff
@@ -721,3 +722,53 @@ impl fmt::Display for TimezoneTimestampTz {
         write!(f, "timezone_{}_tstz", self.0)
     }
 }
+
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize, MzReflect)]
+pub struct ToCharTimestamp {
+    pub format_string: String,
+    pub format: DateTimeFormat,
+}
+
+impl<'a> EagerUnaryFunc<'a> for ToCharTimestamp {
+    type Input = CheckedTimestamp<NaiveDateTime>;
+    type Output = String;
+
+    fn call(&self, input: CheckedTimestamp<NaiveDateTime>) -> String {
+        self.format.render(&*input)
+    }
+
+    fn output_type(&self, input: ColumnType) -> ColumnType {
+        ScalarType::String.nullable(input.nullable)
+    }
+}
+
+impl fmt::Display for ToCharTimestamp {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "tocharts[{}]", self.format_string)
```

Comment:
Nit: use full name `ToCharTimestamp[{}]`. Also in `ToCharTimestampTz`.

Reply:
This is the same name used for the `BinaryFunc` and I'd like to keep them consistent. We could of course change the name of the `BinaryFunc` too, but that would introduce an inconsistency with other `BinaryFunc`s which also use this style of abbreviated names.

Reply:
Yes, a large number of functions currently have the same issue. I think at some point we'll want to make the function names printed in EXPLAIN more easily understandable for users (i.e., closer to SQL), but that's out of scope for this PR.

Reply:
We could possibly start by changing the string for just this function. But resolving this for now.